### PR TITLE
Fix or suppress last cppcheck issues

### PR DESCRIPTION
**Pull request by jrood-nrel**
Let me rephrase my initial comment. I would prefer to somehow solve these, but I think we might just have to suppress them. This should get to 0 warnings though.

### CODE REVIEW

File: amr-wind/core/vs/tensorI.H [hunk 1]
```diff
@@ -153,7 +153,7 @@ operator&&(const TensorT<T>& t1, const TensorT<T>& t2)
 template <typename T>
 AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE T mag_sqr(const TensorT<T>& t)
 {
-    return (t && t);
+    return (t);
```
Comment (sayerhs) [inline amr-wind/core/vs/tensorI.H:156]: This is not correct.

**Review by sayerhs — CHANGES_REQUESTED** [review]
Making sure this doesn't get accidentally merged

File: amr-wind/core/vs/tensorI.H (hunk 1 above)
Comment (jrood-nrel) [reply amr-wind/core/vs/tensorI.H]: Interesting. I just looked this up. I didn't know it was a C++11 thing.

**Review by sayerhs — APPROVED** [review]
